### ISSUE TITLE
Custom Rules can't be reconfigured in a child config

### ISSUE THREAD

**Issue by fredpi** [realm/SwiftLint#3477]
As mentioned by @mgray88, there exists an issue where `custom_rules` can't be reconfigured in a child configuration. E. g. when reconfiguring the severity of a custom rule, one gets two warnings: For the old severity as well as the new severity, instead of just one warning with the reconfigured severity.

Example:

`parent.yml`:

```yaml
custom_rules:
  no_hiding_in_strings:
    regex: "([nN]inja)"
    match_kinds: string
    severity: warning
```

`.swiftlint.yml`:

```yaml
parent_config: parent.yml

custom_rules:
  no_hiding_in_strings:
    regex: "([nN]inja)"
    match_kinds: string
    severity: error
```

Result when running SwiftLint:

```bash
swiftlint
Linting Swift files in current working directory
Linting 'A.swift' (1/1)
FileName:1:21: error: no_hiding_in_strings Violation: Regex matched. (no_hiding_in_strings)
FileName:1:21: warning: no_hiding_in_strings Violation: Regex matched. (no_hiding_in_strings)
Done linting! Found 2 violations, 1 serious in 1 file.
```

[custom-rules-issue.zip](https://github.com/realm/SwiftLint/files/5731460/custom-rules-issue.zip)

SwiftLint Version 0.42.0

**Comment by CraigSiemens** [comment]
It also appears the disabling a custom rule in a child config no longer works as well.

```
disabled_rules:
- my_custom_rule
```

produces this warning
```
warning: 'my_custom_rule' is not a valid rule identifier
```

**Comment by fredpi** [comment]
@CraigSiemens There's a separate issue for that (https://github.com/realm/SwiftLint/issues/3472) and also an active PR addressing it (https://github.com/realm/SwiftLint/pull/3473). Sorry for the inconvenience!

**Comment by fredpi** [comment]
Closed  via https://github.com/realm/SwiftLint/pull/3473.

**Comment by jpsim** [comment]
UnfortunatelyI had to revert #3473 in #3503.

**Comment by jpsim** [comment]
Fixed again in https://github.com/realm/SwiftLint/pull/3538
